### PR TITLE
Update/sidebar banner

### PR DESCRIPTION
**Pull request by adamziel**
PR for a sidebar banner redesigned by @shaunandrews . We want to test different messages in this banner and need to test the new design first.

Old:
old

New:
new

I also added an A/B test to see if reduced contrast isn't causing lower conversion rates.

### CODE REVIEW

**Comment by adamziel** [comment]
Don't review yet please - a few more updates on the way

**Comment by adamziel** [comment]
Okay ready for a review now :-)

File: client/my-sites/current-site/index.jsx [hunk 1]
```diff
@@ -111,14 +111,12 @@ class CurrentSite extends Component {
 		}
 
 		return (
-			<Card className="current-site">
+			<Card className={ this.props.siteCount > 1 ? 'current-site multisite' : 'current-site' }>
```
Comment (dmsnell) [inline client/my-sites/current-site/index.jsx]: also checkout `classNames()`

```js
const cardClasses = classNames( 'current-site', {
	multisite: this.props.siteCount > 1,
} );
```

Comment (adamziel) [reply client/my-sites/current-site/index.jsx]: Good spot! Done done :)

**Comment by klimeryk** [comment]
I've noticed that the domain warnings are now have this margin - is this expected?
screen shot 2018-02-23 at 17 38 07

How it looks like currently on production:
screen shot 2018-02-23 at 17 41 25

I feel like the bottom border on the new upgrade (or probably on the menu below it) and lack of one on top of this nudge (and different background colors) throws off the balance:
screen shot 2018-02-23 at 17 43 20

Adding `Needs Design` label to get some more experienced eyes on this :)

**Comment by adamziel** [comment]
@klimeryk good spot! Looks like an error with updating margin property on notices - I just updated the PR a bit to fix that. @shaunandrews please confirm my latest commit is okay :)

zrzut ekranu 2018-02-26 o 11 30 30

**Comment by marekhrabe** [comment]
Just a quick drive-by comment: the area feels a bit noisy or chaotic to me. We have 6 different horizontal places where things align. Some of them are off just by few px and it might be good to match them to make the area a bit more harmonic. Not saying this belongs to this PR or is a blocker but it definitely caught my eye.

**Comment by shaunandrews** [comment]
I think the SidebarBanner should be "tied" to the site card, at least visually. The notices should "float" in place and stack, just like other notices. Here's what I had in mind:

screen shot 2018-02-26 at 9 29 56 am

This should look pretty much the same on mobile, too.

**Comment by shaunandrews** [comment]
Also, I think we should update the "upgrade and save" notice, too:

screen shot 2018-02-26 at 9 33 09 am

**Comment by adamziel** [comment]
@shaunandrews Okay so the additional margin was purposeful, I see! We can update those other notices, it will require a bit of additional work - we should either update class notice or merge Notice and SidebarBanner components - they seem to be very similar so maybe we don't need to have two separate components. Would you like to proceed with that or should I update this PR?

**Comment by adamziel** [comment]
Actually how about launching A/B test on just the "Free domain with a plan" nudge, and making other design changes a separate project? I think it would give us more meaningful results.

**Comment by shaunandrews** [comment]
> Actually how about launching A/B test on just the "Free domain with a plan" nudge

Sure

**Comment by adamziel** [comment]
@marekhrabe Would you mind confirming if this PR looks good to you? :)

File: client/lib/abtest/active-tests.js [hunk 2]
```diff
@@ -1,5 +1,13 @@
 /** @format */
 export default {
+	redesignedSidebarBanner: {
```
Comment (klimeryk) [inline client/lib/abtest/active-tests.js]: Is this A/B test documented anywhere? As @shaunandrews pointed out, some upsells were not updated, so I wanted to make sure that will be taken into account when analyzing the results of this test.

Best to add this test to the end of the file - that's how the rest is ordered.

File: client/my-sites/current-site/index.jsx [hunk 3]
```diff
@@ -110,15 +111,17 @@ class CurrentSite extends Component {
 			/* eslint-enable wpcalypso/jsx-classname-namespace */
 		}
 
+		const cardClasses = classNames( 'current-site', {
+			multisite: this.props.siteCount > 1,
```
Comment (klimeryk) [inline client/my-sites/current-site/index.jsx]: What's the purpose of this new class? It doesn't seem to be defined anywhere else 🤔

File: client/my-sites/current-site/index.jsx [hunk 4]
```diff
@@ -110,15 +111,17 @@ class CurrentSite extends Component {
...
+			<Card className={ cardClasses }>
 				{ this.props.siteCount > 1 && (
-					<span className="current-site__switch-sites">
-						<Button compact borderless onClick={ this.switchSites }>
-							<Gridicon icon={ rtlOn ? 'arrow-right' : 'arrow-left' } size={ 18 } />
-							{ translate( 'Switch Site' ) }
-						</Button>
-					</span>
+					<Button className="current-site__switch-sites" borderless onClick={ this.switchSites }>
+						<Gridicon icon={ rtlOn ? 'chevron-right' : 'chevron-left' } size={ 24 } />
```
[the remark this answers is not in the document]
Comment (adamziel) [reply client/my-sites/current-site/index.jsx]: @klimeryk I agree, it seems unrelated to the nudge itself; I'll remove all such changes from this PR

[the remark this answers is not in the document]
Comment (adamziel) [reply client/my-sites/current-site/index.jsx]: @klimeryk Also, the post is here: p7EOS0-45A-p2

**Comment by shaunandrews** [comment]
Looks like somethings off with the borders — its a double border on the bottom, and the top border is missing now.
screen shot 2018-02-28 at 9 17 24 am

The text and button could probably stand to be a little bigger on mobile:

screen shot 2018-02-28 at 9 18 05 am

**Comment by adamziel** [comment]
@shaunandrews can you take a look now? I think it will be good :)

**Comment by shaunandrews** [comment]
The borders look better now. 🚢 

This isn't a blocker, but something to think about once your test is done: Maybe we shouldn't show the banner on `/plans/` — it doesn't do anything when you click it there.

**Comment by drw158** [comment]
I like the idea of making the upgrade banners look totally different, so this looks good!

This is late feedback, but it would make sense to me to attach the upgrade banner to the Plan menu item, and the notices would stay attached to the site card.

**Comment by aidvu** [comment]
> Also, I think we should update the "upgrade and save" notice, too:

The `DomainToPaidPlanNotice` already uses the `SidebarBanner`: https://github.com/Automattic/wp-calypso/blob/c70ae2350313211b6a1e5a5c59e73122e01cfd84/client/my-sites/current-site/domain-to-paid-plan-notice.jsx#L49

I believe Igor was just showing different cases. :)

capture

**Comment by aidvu** [comment]
I looked at all of the site notices, and I believe it's a good idea to port them to use `SidebarBanner`. Will make a PR to move the `DomainWarnings` below the site notices. 

I know it's an A/B test, but I want to have it ready if the new ones perform better so we can merge right away. 😆

**Pull request merged**